vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +48,129 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
+                                                        table="csaf_products")
+
+        self._populate_data(csaf_cves, "csaf_products", ["cpe", "package", "module"],
+                            self._get_product_rows, self._product_refresh_cache,
+                            "(%s::int, %s, %s, %s)")
+
+    def _product_refresh_cache(self, cur, delete=False):
+        for product_id, cpe, package, module in cur.fetchall():
+            if delete:
+                del self.csaf_product_map[(cpe, package, module)]
+            else:
+                self.csaf_product_map[(cpe, package, module)] = product_id
+
+    def _get_cve_rows(self, csaf_cves):
+        # TODO: Known_affected are only parsed as of now. Update here accordingly.
+        product_status_id = 4
+
+        to_insert = []
+        to_update = []
+        for cve, csaf_products in csaf_cves.items():
+            for csaf_product in csaf_products:
+                product_id = self.csaf_product_map.get(csaf_product.to_tuple())
+                if not product_id:
+                    self.logger.error(
+                        "CSAF product: %s associated with CVE: %s not found in the database.", csaf_product.to_tuple(), cve)
+                    continue
+                cve_in_cache = self.csaf_cve_map.get((cve, product_id))
+                if not cve_in_cache:
+                    to_insert.append((cve, product_id, product_status_id))
+                    # Insert key to the cache to avoid duplicates.
+                    # CVE ID will be populated after row is inserted into DB.
+                    self.csaf_cve_map[(cve, product_id)] = (-1, product_status_id)
+                else:
+                    cache_product_status_id = cve_in_cache[1]
+                    if product_status_id != cache_product_status_id:
+                        cve_id = cve_in_cache[0]
+                        to_update.append((cve_id, cve, product_id, product_status_id))
+
+        return to_insert, to_update
+
+    def _cve_refresh_cache(self, cur, delete=False):
+        for cve_id, cve, product_id, product_status_id in cur.fetchall():
+            if delete:
+                del self.csaf_cve_map[(cve, product_id)]
+            else:
+                self.csaf_cve_map[(cve, product_id)] = (cve_id, product_status_id)
+
+    def _populate_cves(self, csaf_cves):
+        self.csaf_cve_map = self._prepare_table_map(cols=["cve", "csaf_product_id"],
+                                                    to_cols=["id", "product_status_id"],
+                                                    table="csaf_cves")
+
+        self._populate_data(csaf_cves, "csaf_cves", ["cve", "csaf_product_id", "product_status_id"],
+                            self._get_cve_rows, self._cve_refresh_cache,
+                            "(%s::int, %s, %s::int, %s::int)")
+
+    def _populate_data(self, csaf_cves, table_name, cols, get_rows, refresh_cache, update_tmpl_str):
```

Comment:
this can be done with OOP and inheritance to improve readability as well

Reply:
Many people say the opposite is true.

Reply:
Lets just stick to one style per repository at least. This code is just modified OVAL, something that existed prior to this PR.

Reply:
This code should not be just modified OVAL, this should be the code to store data about CSAF. OVAL contains data per repository and contains all updates and CVEs in a given repo. CSAF is CVE based, it contains information about given CVE and which products it affects. I accept that you were inspired by oval_store implementation, this is a separate module though, we don't need to repeat same mistakes.